src/sorcha/ephemeris/simulation_driver.py

```diff
@@ -137,11 +160,57 @@ def create_ephemeris(args, configs):
                     )
                     outstring += "%lf\n" % (phase_angle * 180 / np.pi)
 
-                    out_csv_file.write(outstring)
+                    out_tuple = (
+                        obj_id,
+                        pointing["FieldID"],
+                        mjd_tai,
+                        jd_tdb,
+                        rho_mag * AU_KM,
+                        drho_magdt * AU_KM / (24 * 60 * 60),
+                        ra0,
+                        dradt * 180 / np.pi,
+                        dec0,
+                        ddecdt * 180 / np.pi,
+                        r_ast_sun[0] * AU_KM,
+                        r_ast_sun[1] * AU_KM,
+                        r_ast_sun[2] * AU_KM,
+                        v_ast_sun[0] * AU_KM / (24 * 60 * 60),
+                        v_ast_sun[1] * AU_KM / (24 * 60 * 60),
+                        v_ast_sun[2] * AU_KM / (24 * 60 * 60),
+                        obs_sun[0] * AU_KM,
+                        obs_sun[1] * AU_KM,
+                        obs_sun[2] * AU_KM,
+                        dobs_sundt[0] * AU_KM / (24 * 60 * 60),
+                        dobs_sundt[1] * AU_KM / (24 * 60 * 60),
+                        dobs_sundt[2] * AU_KM / (24 * 60 * 60),
+                        phase_angle * 180 / np.pi,
+                    )
+
+                    in_memory_csv.writerow(out_tuple)
+
+    # reset to the beginning of the in-memory CSV
+    output.seek(0)
```

Comment:
when I initially saw this I assumed that writing in memory and then reading the in mem csv to a dataframe would be slower than just adding a row to an existing dataframe, but I ran the numbers and it's actually faster by a factor of ~27 🤯

Reply:
Yes, my understanding here is that any time you append to a dataframe a new copy of the data structure has to be made, so with small dataframes it's not a big deal, but as the size increases it becomes more efficient to append to a csv, then convert to a df.